src/digidoc-tool.cpp

```diff
@@ -830,6 +844,51 @@ static int create(int argc, char* argv[])
 
 
 /**
+ * Create new container.
+ *
+ * @param argc number of command line arguments.
+ * @param argv command line arguments.
+ * @return EXIT_FAILURE (1) - failure, EXIT_SUCCESS (0) - success
+ */
+static int createBatch(int argc, char* argv[])
+{
+    Params p(argc, argv);
+    if(p.path.empty())
+    {
+        printUsage(argv[0]);
+        return EXIT_FAILURE;
+    }
+
+    unique_ptr<Signer> signer;
+    try {
+        signer = getSigner(p);
+    } catch(const Exception &e) {
+        parseException(e, "Caught Exception:");
+        return EXIT_FAILURE;
+    }
+
+    int returnCode = EXIT_SUCCESS;
+    for(const string &file: File::listFiles(p.path))
+    {
+        if(file.compare(file.size() - 6, 6, ".asice") == 0)
+            continue;
+        std::cout << "Signing file: " << file << endl;
+        try {
+            unique_ptr<Container> doc(Container::create(file + ".asice"));
+            doc->addDataFile(file, "application/octet-stream");
+            if(signContainer(doc.get(), signer.get(), p.dontValidate) == EXIT_FAILURE)
```

Comment:
Default is int returnCode = EXIT_SUCCESS;
and turn total status failure only then when one if(signContainer == EXIT_FAILURE)" fails.
It does not terminate process when some signatures are invalid